### PR TITLE
DM-2369: Fix site and practice show visit counts

### PR DESCRIPTION
**Pull request by emyl3**
### JIRA issue link
[DM-2369](https://agile6.atlassian.net/secure/RapidBoard.jspa?rapidView=22&projectKey=DM&modal=detail&selectedIssue=DM-2369)

## Description - what does this code do?
This PR allows for the following:
- not double count "Practice show" visits when clicking on a practice's practice card
- filter site visits by the `is_duplicate` property so we do not count those visits that were corrected by this [script](https://agile6.atlassian.net/browse/DM-2375). (**NOTE**: The way we are currently fetching "Practice show" counts on the metrics page -- we may want to refactor later to use the same queries throughout the codebase--meant we did not need to explicitly filter by the `is_duplicate` property added by this [script](https://agile6.atlassian.net/browse/DM-2374).) We needed to filter by the is_duplicate property when getting practice views on the admin dashboard 
- refactor the "last updated" message so that we use a helper to display the session's username and use the `local_time` gem to display the time

## Testing done - how did you test it/steps on how can another person can test it 
**Practice show double count**
1. Do not log in
2. Click on a practice card from the homepage (highlighted practice's Practice card)
3. Check you ahoy_events table and ensure no "Practice show" count was added when you were not logged in
4. Now, log in as a user
5. Click on a practice card
6. Check you ahoy_events table and ensure only one "Practice show" count was added for that practice
7. Navigate to another practice show page via changing the url
8. Check you ahoy_events table and ensure only one "Practice show" count was added for that practice

**Practice show count**
1. Go to a practice's metrics page and scroll to the leaderboard section
2. Click on the all-time tab
3. Take note of the number of views for the practice leaderboards
4. Now run [this script](https://agile6.atlassian.net/browse/DM-2376) for a certain month and year and take note of how many practices were updated
5. Refresh the practice leaderboards and make sure the amount of views for the practices went down accordingly (NOTE: if the practice is not on the leaderboard but it's count was updated this may not reflect on the leaderboard so keep that in mind)
6. Go to the admin dashboard and ensure the same practice view counts in the "Practice leaderboards" tab

**Site visit count**
1. Go to the admin panel and take note of the number of "Page Views from last month"
2. Run [this script](https://agile6.atlassian.net/browse/DM-2375) for last month and take note of the number updated
3. Refresh the admin panel and the "Page Views from last month" should be a different number that takes into account the number updated in the script

**Last updated message**
1. Go to a practice that had an expired and ended session
2. Check that the last updated message is in localtime and the user who last updated it is displaying correctly
3. Try to edit a practice another editor is editing
4. Make sure the message when you are redirected to the Metrics page is displaying correctly with the username and if they are a Site admin or not

**Updated**
1. Make sure comments show up on the Practice show page
2. Go to practice via practice card, ensure only one "Practice show" visit
3. Now click on a non-practice show page like return to homepage or covid-19 page, there should not be another "Practice show" visit in the DB
4. Navigate to a practice show page via URL and there should only be one "Practice show" visit in the DB

## Acceptance criteria
- [ ]

## Definition of done
- [X] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/views/layouts/_ahoy_event_tracking.js.erb [hunk 1]
```diff
@@ -44,7 +43,5 @@ document.addEventListener('turbolinks:load', function () {
     if (shouldTrackVisit(visitProps, visitorProps)) {
       trackVisit(visitProps, visitorProps);
     }
-  } else {
```
Comment (emyl3) [inline app/views/layouts/_ahoy_event_tracking.js.erb:47]: ⚠️ removed because I don't this would ever be hit

File: app/models/practice.rb [hunk 2]
```diff
@@ -326,9 +330,6 @@ def number_of_unsuccessful_adoptions
     diffusion_histories.joins(:diffusion_history_statuses).where(diffusion_history_statuses: {status: 'Unsuccessful'}).count
   end
 
-  def date_range_views(start_date, end_date)
```
Comment (emyl3) [inline app/models/practice.rb:329]: ⚠️ Just moved it closer with the other view count code for readability

File: app/views/practices/show/authenticated/_show_authenticated.html.erb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 <% provide :head_tags do %>
   <%= javascript_include_tag 'practice_page', 'data-turbolinks-track': 'reload' %>
+  <%= javascript_tag 'data-turbolinks-track': 'reload' do %>
```
Comment (emyl3) [inline app/views/practices/show/authenticated/_show_authenticated.html.erb:3]: ⚠️ Only want to track the "Practice show" when the user is logged in so only included in the authenticated view

File: app/models/practice.rb [hunk 4]
```diff
@@ -89,7 +89,11 @@ def has_facility?
 
   # views
   def views
-    Ahoy::Event.where_props(practice_id: id).count
+    Ahoy::Event.where(name: 'Practice show').where_props(practice_id: id).where("properties->>'is_duplicate' is null").count
```
Comment (emyl3) [inline app/models/practice.rb:92]: ⚠️ just explicitly calling out it is the "Practice show" event here as well as filtering for `is_duplicate` `nil`

**Review by JoshingYou1 — APPROVED** [review]
I see the comment section! :smiley:  Everything else still works as intended. Nice find and great work Elisa!

**Review by bradjohnson92008 — APPROVED** [review]
I was able to get through all the tests successfully - great job on this Elisa!

**Pull request merged**